lib/scenic/adapters/postgres/views.rb

```diff
@@ -0,0 +1,56 @@
+module Scenic
+  module Adapters
+    class Postgres
+      # Fetches defined views from the postgres connection.
+      # @api private
+      class Views
+        def initialize(connection)
+          @connection = connection
+        end
+
+        # All of the views that this connection has defined.
+        #
+        # This will include materialized views if those are supported by the
+        # connection.
+        #
+        # @return [Array<Scenic::View>]
+        def all
+          views + materialized_views
+        end
+
+        private
+
+        attr_reader :connection
+
+        def views
+          connection.execute(<<-SQL).map { |result| to_scenic_view(result) }
```

Comment:
I'm not a huge fan of putting something other than `strip` or `strip_heredoc` after the heredoc declaration.

What about having `#views` and `#materialized_views` return the raw database arrays, and have `#all` convert them with something like `to_scenic_views(views + materialized_views)`?